### PR TITLE
[DDW-354] Merge release/0.11.0 into develop branch

### PR DESCRIPTION
**Pull request by nikolaglumac**
This PR merges `release/0.11.0` branch into `develop`  and fixes all the conflicts.
In the scope of this PR `reportRequest` was updated and correctly flow-typed (it was missing the `// @flow` declaration!

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@DominikGuzei please review this PR. It contains one very important change: https://github.com/input-output-hk/daedalus/pull/1011/files#diff-dd0680109c32c035ae8b95d8de8ab4b6
(please check out the flow type declarations here and also in the related `sendAdaBugReport` and `sendEtcBugReport` JS files)
Thanks!

**Comment by nikolaglumac** [comment]
@rvl please review DEVOPS changes and please don't merge this PR until @DominikGuzei reviews the other part ;)

File: source/renderer/app/api/lib/reportRequest.js [hunk 1]
```diff
@@ -28,9 +30,9 @@ export type RequestPayload = {
   }
 };
 
-function typedHttpRequest<Response>(
+function typedHttpRequest(
   httpOptions: RequestOptions, requestPayload?: RequestPayload
-): Promise<Response> {
+): Promise<void> {
```
Comment (DominikGuzei) [inline source/renderer/app/api/lib/reportRequest.js]: Why did you remove the response types here?

File: source/renderer/app/api/etc/sendEtcBugReport.js [hunk 2]
```diff
@@ -16,35 +16,22 @@ export type SendEtcBugReportRequestParams = {
 
 export const sendEtcBugReport = (
   { requestFormData, application }: SendEtcBugReportRequestParams
-): Promise<{}> => {
+): Promise<void> => {
```
Comment (DominikGuzei) [inline source/renderer/app/api/etc/sendEtcBugReport.js]: This basically circumvents typing (except of the fact that we are returning a promise) … shouldn't we define the response type here?

Comment (nikolaglumac) [reply source/renderer/app/api/etc/sendEtcBugReport.js]: Because we are actually not returning anything!
I am not sure if this is right - this was required for both lint and flow:type to run successfully...

Comment (DominikGuzei) [reply source/renderer/app/api/etc/sendEtcBugReport.js]: We are returning the promise that is created by the typed `request` method 😉 so the type you define here gets passed down to that request method so that flow knows that this is what we expect to come back as a result from the server side. But im not sure what the server actually returns (as we don't have type information about this yet)
